mcm/tools/installer.py

```diff
@@ -54,10 +52,77 @@ def close(self):
         if self.cleanup:
             try:
                 self.logger.error('Deleting the directory: %s' % self.directory)
-                shutil.rmtree(self.directory)
+                self._remove_folder(self.directory)
             except Exception as ex:
                 self.logger.error('Could not delete directory "%s". Reason: %s' % (self.directory, ex))
 
+    def _folder_exists(self, path):
+        """
+        Checks if a folder exists using the remote session.
+
+        Arguments:
+            path (str): Absolute path to check.
+
+        Returns:
+            bool: True if the folder exists, False otherwise.
+        """
+        _, stdout, stderr = self.ssh_session.execute("ls -l '%s'" % (path))
```

Comment:
Open an SFTP session and use [listdir()](https://docs.paramiko.org/en/latest/api/sftp.html#paramiko.sftp_client.SFTPClient.listdir):
```py
self.ssh_client.open_sftp().listdir(path)
```